File: services/ai-storage-service/src/services/aiProviders/groq.js
const fetch = require('node-fetch');
const env = require('../../config/env');

// Groq exposes an OpenAI-compatible chat completions endpoint.
async function chat({ messages, model }) {
  const response = await fetch('https://api.groq.com/openai/v1/chat/completions', {
    method: 'POST',
    headers: { 'content-type': 'application/json', authorization: `Bearer ${env.ai.apiKey}` },
    body: JSON.stringify({ model: model || env.ai.model, messages }),
  });
  if (!response.ok) {
    const text = await response.text();
    throw Object.assign(new Error(`Groq API error: ${text}`), { status: 502 });
  }
  const data = await response.json();
  return {
    content: data.choices?.[0]?.message?.content || '',
    usage: data.usage || {},
  };
}

module.exports = { chat };

File: services/ai-storage-service-python/services/ai_providers/groq.py
import httpx
from config.settings import settings
from loguru import logger
from typing import Dict, Any


async def chat(messages: list[Dict[str, str]], model: str = None) -> Dict[str, Any]:
    url = "https://api.groq.com/openai/v1/chat/completions"
    headers = {
        "content-type": "application/json",
        "authorization": f"Bearer {settings.ai_api_key}"
    }
    payload = {
        "model": model or settings.ai_model,
        "messages": messages
    }
    
    async with httpx.AsyncClient() as client:
        response = await client.post(url, headers=headers, json=payload)
        
    if not response.is_success:
        error_text = response.text
        logger.error(f"Groq API error: {error_text}")
        raise Exception(f"Groq API error: {error_text}")
    
    data = response.json()
    return {
        "content": data.get("choices", [{}])[0].get("message", {}).get("content", ""),
        "usage": data.get("usage", {})
    }
